modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/NameSign.java

```diff
@@ -15,49 +14,50 @@
+
+    /**
+     * When a craft is piloted, check the craft for a name sign
+     */
     @EventHandler
     public void onCraftDetect(@NotNull CraftDetectEvent event) {
-        Craft c = event.getCraft();
+        Craft craft = event.getCraft();
 
-        if(c.getNotificationPlayer() == null || (Settings.RequireNamePerm && !c.getNotificationPlayer().hasPermission("movecraft.name.use"))) {
-            //Player is null or does not have permission (when required)
+        //Ensure nonnull player and sufficient permissions
+        if (craft.getNotificationPlayer() == null || (Settings.RequireNamePerm && !craft.getNotificationPlayer().hasPermission("movecraft.name.use"))) {
             return;
         }
 
-        World w = c.getW();
+        World craftWorld = craft.getW();
+        for (MovecraftLocation location : craft.getHitBox()) { //Iterate through every craft block to search for name sign
+            Block b = location.toBukkit(craftWorld).getBlock();
+            if (b.getType() == Material.SIGN_POST || b.getType() == Material.WALL_SIGN) { //If block is a sign
+                Sign sign = (Sign) b.getState();
+                String craftNameFromSign = "";
 
-        for (MovecraftLocation location : c.getHitBox()) {
-            Block b = location.toBukkit(w).getBlock();
-            if (b.getType() == Material.SIGN_POST || b.getType() == Material.WALL_SIGN) {
-                Sign s = (Sign) b.getState();
-                String name = "";
-                if (s.getLine(0).equalsIgnoreCase(HEADER)) {
-                    boolean firstName = true;
-                    for (int i = 1; i <= 3; i++) {
-                        if (s.getLine(i) != "") {
-                            if (firstName) {
-                                firstName = true;
-                            } else {
-                                name += " ";
-                                //Add a space between lines for all after the first.
-                            }
-                            name += s.getLine(i);
+                if (sign.getLine(0).equalsIgnoreCase(NAME_SIGN_IDENTIFIER)) { //If name sign is found
+                    for (int i = 1; i <= 3; i++) { //Go through each line below the NAME_SIGN_IDENTIFIER string
+                        if (!sign.getLine(i).isEmpty()) { //If the line is not empty
+                            craftNameFromSign += sign.getLine(i) + " "; //Add the line to the name, with a space at the end
                         }
                     }
-                    c.setName(name);
+                    craft.setName(craftNameFromSign.isEmpty() ? "" : craftNameFromSign.substring(0, craftNameFromSign.length() - 1)); //Set craft name if the sign wasn't blank, while removing the last space
                     return;
                 }
             }
         }
     }
+
+    /**
+     * When player is creating a new Name Sign
+     */
```

Comment:
I would advise removing this comment - methods intended for internal use should typically should not use javadoc comments. While event handlers are expose as public methods, they do so for a matter of compatibility with spigot rather than for users of the movecraft API.